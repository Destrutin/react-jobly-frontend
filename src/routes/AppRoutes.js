import React, {useContext} from "react";
import {Routes, Route, Navigate} from "react-router-dom";
import Homepage from "../homepage/Homepage";
import CompanyList from "../companies/CompanyList";
import CompanyDetails from "../companies/CompanyDetails";
import JobList from "../jobs/JobList";
import LoginForm from "../auth/LoginForm";
import SignupForm from "../auth/SignupForm";
import ProfileForm from "../profiles/ProfileForm";
import UserContext from "../auth/UserContext";

function AppRoutes({signup, login}) {
    const {currentUser} = useContext(UserContext);
    return (
        <>
            <Routes>
                {!currentUser &&
                <>
                    <Route 
                        exact path="/login" 
                        element={<LoginForm login={login}/>}
                    />

                    <Route 
                        exact path="/signup" 
                        element={<SignupForm signup={signup}/>}
                    />
                </>
                }

                <Route 
                    exact path="/" 
                    element={<Homepage/>}
                />

                {currentUser &&
                <>
                    <Route 
                        exact path="/companies" 
                        element={<CompanyList/>}
                    />

                    <Route 
                        exact path="/companies/:handle" 
                        element={<CompanyDetails/>}
                    />

                    <Route 
                        exact path="/jobs" 
                        element={<JobList/>}
                    />

                    <Route 
                        exact path="/profile" 
                        element={<ProfileForm/>}
                    />
                </>
                }
                

                <Route path="/" element={<Navigate to="/"/>}/>
            </Routes>
        </>
    )
}


export default AppRoutes;