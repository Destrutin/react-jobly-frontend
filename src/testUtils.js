import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  password: "omgsecret",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
};

const UserProvider =
    ({ children, currentUser = demoUser, hasAppliedToJob = () => false }) => (
    <UserContext.Provider value={{ currentUser, hasAppliedToJob }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };
