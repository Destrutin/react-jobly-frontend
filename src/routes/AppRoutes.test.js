import React from "react";
import {render} from "@testing-library/react";
import {MemoryRouter} from "react-router";
import {UserProvider} from "../testUtils";
import AppRoutes from "./AppRoutes";

it("renders without crashing", function () {
  render(
      <MemoryRouter>
        <UserProvider>
          <AppRoutes/>
        </UserProvider>
      </MemoryRouter>,
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <AppRoutes/>
        </UserProvider>
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
