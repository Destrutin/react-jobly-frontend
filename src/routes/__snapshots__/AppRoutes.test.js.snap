// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Homepage"
  >
    <header
      class="homepage-header"
    >
      <h1>
        Jobly
      </h1>
      <p>
        All the jobs in one, convenient place.
      </p>
    </header>
    <section
      class="homepage-section"
    >
      <h2>
        Welcome Back, !
      </h2>
    </section>
  </div>
</DocumentFragment>
`;
